core/trino-main/src/main/java/io/trino/operator/TopNProcessor.java

```diff
@@ -78,28 +87,32 @@ public Page getOutput()
     {
         if (outputIterator == null) {
             // start flushing
-            outputIterator = topNBuilder.buildResult();
+            outputIterator = topNBuilder == null ? emptyIterator() : topNBuilder.buildResult();
         }
 
         Page output = null;
         if (outputIterator.hasNext()) {
             output = outputIterator.next();
         }
         else {
-            outputIterator = emptyIterator();
+            outputIterator = null;
+            topNBuilder = null;
```

Comment:
After you build the result on line 90, it's not clear that any of the other side effects happening to topNBuilder will be reflected in the output (even though it may not occur in practice), it just looks more correct to an average programmer to clear the topNBuilder immediately after calling buildResult, so we should do that (unless it makes this code incorrect somehow).

Reply:
Even though buildResult() is called, topNBuilder is still maintained in outputIterator. So, it needs to be memory-accounted until outputIterator is nullified. This is the reason why other operators like HashAggregationOperator and TopNRankingOperator maintain their builder until flushing is completed.